controllers/serviceexport/serviceexport.go

```diff
@@ -242,13 +242,18 @@ func (r *Reconciler) ReconcileAliases(
 	// Check if an update is needed to the alias info in the status
 	updateNeeded := false
 
-	// Status does not contain alias info. Could be the first iteration of the reconcile loop
-	if len(serviceexport.Status.Aliases) == 0 {
-		updateNeeded = true
-	}
+	// There is no do..while in Go, using the 'for' loop to achieve do..while here. It helps to keep the code
+	// clean in cases like these where the control flow needs to jump (like the goto statement in C) to an 
+	// instruction deeper down the function when a condition is satisfied and there is no need to test any 
+	// other conditions.
+	for doOnce := true; doOnce; doOnce = false {
```

Comment:
If I understood correctly, you re running the for loop only once.
1. Why do you need for loop exactly?
2. Why do you need break statement?